script/testing/oltpbench/run_oltpbench_v2.py

```diff
@@ -0,0 +1,121 @@
+            if oltp_testcase_loop:
+                # also support the combination of differetn loop options
+                loop_dict = {}
+                for loop_key in oltp_testcase_loop.keys():
+                    loop_dict[loop_key] = oltp_testcase_loop[loop_key]
+
+                keys, values = zip(*loop_dict.items())
+                loop_combination_dicts = [
+                    dict(zip(keys, v)) for v in itertools.product(*values)]
+
+                for loop_item in loop_combination_dicts:
+                    for loop_key in loop_item.keys():
+                        oltp_testcase_base[loop_key] = loop_item[loop_key]
+                    oltp_test_suite.append(TestCaseOLTP(oltp_testcase_base))
+            else:
+                # there is no loop 
+                oltp_test_suite.append(TestCaseOLTP(oltp_testcase_base))
+    return args, oltp_test_suite
+
+
+def parse_parameters(aparser, args):
+    aparser.add_argument("benchmark", help="Benchmark Type")
+    aparser.add_argument("weights", help="Benchmark weights")
+    aparser.add_argument("--db-host", help="DB Hostname")
+    aparser.add_argument("--db-port", type=int, help="DB Port")
+    aparser.add_argument("--db-output-file", help="DB output log file")
+
+    aparser.add_argument("--scale-factor", type=float, metavar="S",
+                         help="The scale factor. (default: 1)")
+    aparser.add_argument("--transaction-isolation", metavar="I",
+                         help="The transaction isolation level (default: TRANSACTION_SERIALIZABLE")
+    aparser.add_argument("--client-time", type=int, metavar="C",
+                         help="How long to execute each benchmark trial (default: 20)")
+    aparser.add_argument("--terminals", type=int, metavar="T",
+                         help="Number of terminals in each benchmark trial (default: 1)")
+    aparser.add_argument("--loader-threads", type=int, metavar="L",
+                         help="Number of loader threads to use (default: 1)")
+    aparser.add_argument("--build-type",
+                         default="debug",
+                         choices=["debug", "release", "relwithdebinfo"],
+                         help="Build type (default: %(default)s")
+    aparser.add_argument("--query-mode",
+                         default="simple",
+                         choices=["simple", "extended"],
+                         help="Query protocol mode")
+
+    args = vars(aparser.parse_args(args))
+
+    oltp_testcase = TestCaseOLTP(args)
+
+    return args, [oltp_testcase]
+
+
+if __name__ == "__main__":
+
+    args = sys.argv[1:]
+    aparser = argparse.ArgumentParser(description="Timeseries")
+    test_suite = []
+
+    if args and args[0].endswith(".json"):
+        # trace file mode
+        args, test_suite = parse_tracefile(aparser, args)
+    else:
+        # parameter mode
+        args, test_suite = parse_parameters(aparser, args)
```

Comment:
it is like a binary choice, either config file mode or command line parameter mode. I know it is a pure design decision, but do we want to support hybrid parameter loading with prioritization as commented above?

Reply:
no, we only support the config file approaches